test/Table_Tests/src/In_Memory/Split_Tokenize_Spec.enso

```diff
@@ -183,32 +183,75 @@ spec =
             Problems.test_problem_handling action problems tester
 
         Test.specify "tokenize will make column names unique" <|
-            cols = [["foo", [0, 1, 2]], ["bar", ["a12b34r5", "23", "2r4r55"]], ["bar_1", ["a", "b", "c"]]]
+            cols = [["foo", [0, 1, 2]], ["bar", ["a12b34r5", "23", "2r4r55"]], ["bar 1", ["a", "b", "c"]]]
             t = Table.new cols
             expected_rows = [[0, "12", "34", "5", "a"], [1, "23", Nothing, Nothing, "b"], [2, "2", "4", "55", "c"]]
-            expected = Table.from_rows ["foo", "bar_0", "bar_1_1", "bar_2", "bar_1"] expected_rows
+            expected = Table.from_rows ["foo", "bar 0", "bar 1_1", "bar 2", "bar 1"] expected_rows
             action = t.tokenize_to_columns "bar" "\d+"  on_problems=_
             tester = t-> t.should_equal_verbose expected
-            problems = [Duplicate_Output_Column_Names.Error ["bar_1"]]
+            problems = [Duplicate_Output_Column_Names.Error ["bar 1"]]
             Problems.test_problem_handling action problems tester
 
     Test.group "Table.split/tokenize column order" <|
         Test.specify "preserves column order" <|
             cols = [["foo", [0, 1, 2]], ["bar", ["abc", "cbdbef", "ghbijbu"]], ["baz", [1, 2, 3]]]
             t = Table.new cols
             expected_rows = [[0, "a", "c", Nothing, 1], [1, "c", "d", "ef", 2], [2, "gh", "ij", "u", 3]]
-            expected = Table.from_rows ["foo", "bar_0", "bar_1", "bar_2", "baz"] expected_rows
+            expected = Table.from_rows ["foo", "bar 0", "bar 1", "bar 2", "baz"] expected_rows
             t2 = t.split_to_columns "bar" "b"
             t2.should_equal_verbose expected
 
+    Test.group "Table.parse_to_columns" <|
+        Test.specify "can parse to columns" <|
+            t = Table.from_rows ["foo", "bar", "baz"] [["x", "12 34p q56", "y"], ["xx", "a48 59b", "yy"]]
+            expected = Table.from_rows ["foo", "bar 0", "bar 1", "baz"] [["x", 1, 2, "y"], ["x", 3, 4, "y"], ["x", 5, 6, "y"], ["xx", 4, 8, "yy"], ["xx", 5, 9, "yy"]]
+            actual = t.parse_to_columns "bar" "(\d)(\d)"
+            actual.should_equal_verbose expected
+
+        Test.specify "no regex groups" <|
+            t = Table.from_rows ["foo", "bar", "baz"] [["x", "12 34p q56", "y"], ["xx", "a48 59b", "yy"]]
+            expected = Table.from_rows ["foo", "bar", "baz"] [["x", 12, "y"], ["x", 34, "y"], ["x", 56, "y"], ["xx", 48, "yy"], ["xx", 59, "yy"]]
+            actual = t.parse_to_columns "bar" "\d\d"
+            actual.should_equal_verbose expected
+
+        Test.specify "named groups" <|
+            t = Table.from_rows ["foo", "bar", "baz"] [["x", "12 34p q56", "y"], ["xx", "a48 59b", "yy"]]
+            expected = Table.from_rows ["foo", "xomt", "biff", "baz"] [["x", 1, 2, "y"], ["x", 3, 4, "y"], ["x", 5, 6, "y"], ["xx", 4, 8, "yy"], ["xx", 5, 9, "yy"]]
+            actual = t.parse_to_columns "bar" "(?<xomt>\d)(?<biff>\d)"
+            actual.should_equal_verbose expected
```

Comment:
I'd appreciate two more tests:

- starting table has columns `foo`, `bar` and `bar 1`, we parse `bar` with a regex that has 2 groups. We will have a duplicate `bar 1` name - how will this behave?
- starting table has columns `foo`, `bar`, `baz`, we parse `bar` with a regex that has 3 named groups: `bar`, `baz`, `quux`.

Reply:
And after looking at the code I feel like we should have a few more:
- a table with 3 columns `foo`, `bar` and `baz`, but **no rows**, parsed with Regexes `\d+` and `(\d)(\d)`. I feel like the current behaviour may not necessarily be correct. I think with this operation the structure of the output table should depend only on what columns it has and what is the Regex - but not on the rows. So IMO these two tests should return tables with no rows, but with columns `['foo', 'bar', 'baz']` and `['foo', 'bar 1', 'bar 2', 'baz']` each. Do you agree?
- as above, but there are rows - just the first one has no matches - because it is `""` empty or `Nothing`, but there are latter rows that do have matches.

Reply:
Agree with all of this. I got rid of the column namer, and instead I'm just passing in the column names; I don't need the generality of a namer. However, I wasn't able to add tests for tables with no rows, because it could not infer that the empty columns were text columns. I did the equivalent for tables with no matches.

Reply:
Is there a way to set the type of an empty column?

Reply:
Yes, currently it is indeed a bit problematic, we have plans to improve on this: #6159 

But it is possible. My standard trick is to create a table with a single row, so that it will set the types correctly and then truncate it by filtering:
```
t0 = table_builder [["X", ["x"]]]
# This will keep only rows that contain `Nothing`, but there no such rows in `t0`, so it will have no rows.
t = t0.filter "X" Filter_Condition.Is_Nothing
```

Reply:
Oh, if you are working in just in-memory, it's even easier - just `take 0` rows:
```
t0 = Table.new [["X", ["x"]]]
t = t0.take 0
```